x/btcstaking/types/btc_slashing_tx.go

```diff
@@ -274,6 +274,7 @@ func (tx *BTCSlashingTx) BuildSlashingTxWithWitness(
 	outputIdx uint32,
 	delegatorSig *bbn.BIP340Signature,
 	covenantSigs []*asig.AdaptorSignature,
+	covenantQuorum uint32,
```

Comment:
maybe we can return error early from this function i.e:
```
if len(covenantSigs) < covenantQuorum {
  return nil, fmt.Errorf("not enough signatures")
}
```

Reply:
Good point! As `covenantSigs` can contain `nil` (in order to have the length of covenant committee size), I added the below logic below L303

```go
	// ensure the number of covenant signatures is at least the quorum number
	if numSigs < covenantQuorum {
		return nil, fmt.Errorf("not enough covenant signatures to reach quorum")
	}
```

does this make sense?